app/services/book_sorting_service.rb

```diff
@@ -0,0 +1,33 @@
+class BookSortingService
+  SORT_TITLES = {
+    date_asc: I18n.t('newest_first'),
+    popular: I18n.t('popular_first'),
+    price_asc: I18n.t('price_asc'),
+    price_desc: I18n.t('price_desc'),
+    title_asc: I18n.t('title_asc'),
+    title_desc: I18n.t('title_desc')
+  }.freeze
+
+  def initialize(params)
+    @sort_type = params[:sort]&.to_sym
+  end
+
+  def sort(books)
+    @given_books = books || Book.all
+    case @sort_type
+    when :date_asc then @given_books.order('created_at asc')
```

Comment:
rewrite this logic
you can simply pass such string into params[:sort]
`created_at asc` or `price asc`
after do
`params[:sort].split(' ')`
For most popular you can pass
`most_popular asc`
and create scope `most_popular` inside books model
scope can receive attribute `asc/desc`

So to summarize it will be better to define scopes inside model for each sort case and just pass
`params[:sort] = '<scope_name> <asc/desc>'`